src/hash.rs

```diff
@@ -0,0 +1,143 @@
+use Error;
+use ffi;
+use std::ffi::CStr;
+use std::mem;
+
+
+/// A hash function.
+#[derive(Clone, Copy, Eq, PartialEq)]
+pub struct Hash {
+    raw: *const ffi::ltc_hash_descriptor,
+}
+
+impl Hash {
+    /// Get the name of the hash function.
+    pub fn name(&self) -> &str {
+        unsafe {
+            CStr::from_ptr((*self.raw).name).to_str().unwrap()
+        }
+    }
+
+    /// Get the digest output size in bytes.
+    pub fn size(&self) -> u64 {
+        unsafe {
+            (*self.raw).hashsize
+        }
+    }
+
+    /// Get the block size the hash uses.
+    pub fn block_size(&self) -> u64 {
+        unsafe {
+            (*self.raw).blocksize
+        }
+    }
+
+    /// Convenience function to compute the hash of `input`.
+    pub fn hash<I: AsRef<[u8]>>(&self, input: I) -> Result<Vec<u8>, Error> {
+        let mut state = HashState::new(*self);
```

Comment:
Is there a reason to make `I` generic? I'm just wondering because all other methods take `&[u8]`.

Reply:
Mostly for convenience (since this is a convenience method anyway) so that you can pass in a `&str` or other things, e.g. `MD5.hash("foo")`.